packages/expo-modules-core/ios/AppDelegates/EXAppDelegateWrapper.h

```diff
@@ -6,10 +6,24 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+@interface RCTAppDelegate ()
+
+- (RCTRootViewFactory *)createRCTRootViewFactory;
+
+@end
+
+__deprecated_msg("EXAppDelegateWrapper is deprecated as of SDK 53. Migrate your AppDelegate to Swift and use ExpoAppDelegate instead.")
```

Comment:
could we show the warning when sdk >= 53? i guess we will later cherry-pick this to sdk-52 to support 0.77, but then people stick on sdk 52 and 0.76 will see the warning.

Reply:
On one side I agree, but on the other I feel like people could already migrate to `ExpoAppDelegate` once this change is out. Then I would only change the message that it's deprecated as of SDK 52. What do you think?

Reply:
> sdk52 CNG project would have the warning

Why? It would use the updated project template, right? Even if we use the old template, is that really a problem? If we publish a blog post about support for 0.77, I think it is a good idea to include a note about it too.
I'd just prefer having this change earlier, so we can drop support for the wrapper earlier.

Reply:
in my opinion, we should not introduce breaking changes in sdk 52 lifecycle even to support rn 0.77. that's why i introduced the config-plugin: https://github.com/expo/expo/blob/main/packages/%40expo/prebuild-config/src/plugins/sdk52/ReactNative77CompatPlugin.ts
if we change AppDelegate.swift in the middle of sdk 52, that may break people's config-plugin when they have dangerous plugins.

though, it's not a big deal and is a warning only. i didn't have strong opinion here and can keep the code as-is.